nixos/modules/misc/documentation.nix

```diff
@@ -109,6 +109,7 @@ let
         } ''
           export NIX_STORE_DIR=$TMPDIR/store
           export NIX_STATE_DIR=$TMPDIR/state
+          export NIX_ABORT_ON_WARN=true
```

Comment:
Given the [`manual-nixos.yml`](https://github.com/NixOS/nixpkgs/blob/6c0a5bef965318da037febf4114e4dfe52ada460/.github/workflows/manual-nixos.yml#L31) workflow uses non-flake CLI and the whole motivation behind this PR is improving the `manual-nixos.yml` workflow to be more useful, I lean towards simply adding `NIX_ABORT_ON_WARN` to `impureEnvVars` instead of adding a nixos options for it.

Thanks for your review!

Reply:
Wait. We already have `documentation.nixos.options.warningsAreErrors` that defaults to true :eyes:

I'll look some more into this. Might be a regression.